tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs

```diff
@@ -0,0 +1,188 @@
+            var saslContinueReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                "{conversationId: 1, done: true, payload: BinData(0,\"\"), ok: 1}"));
+
+            var subject = new MongoAWSAuthenticator(credential, null, randomByteGeneratorMock.Object, dateTimeProviderMock.Object);
+
+            var connection = new MockConnection(serverId);
+            connection.EnqueueReplyMessage(saslStartReply);
+            connection.EnqueueReplyMessage(saslContinueReply);
+
+            var expectedRequestId = RequestMessage.CurrentGlobalRequestId + 1;
+
+            Exception exception;
+            if (async)
+            {
+                exception = Record.Exception(() => subject.AuthenticateAsync(connection, connectionDescription, CancellationToken.None).GetAwaiter().GetResult());
+            }
+            else
+            {
+                exception = Record.Exception(() => subject.Authenticate(connection, connectionDescription, CancellationToken.None));
+            }
+            exception.Should().BeNull();
+
+            SpinWait.SpinUntil(() => connection.GetSentMessages().Count >= 2, TimeSpan.FromSeconds(5)).Should().BeTrue();
+
+            var sentMessages = MessageHelper.TranslateMessagesToBsonDocuments(connection.GetSentMessages());
+            sentMessages.Count.Should().Be(2);
+
+            var actualRequestId0 = sentMessages[0]["requestId"].AsInt32;
+            var actualRequestId1 = sentMessages[1]["requestId"].AsInt32;
+            actualRequestId0.Should().BeInRange(expectedRequestId, expectedRequestId + 10);
+            actualRequestId1.Should().BeInRange(actualRequestId0 + 1, actualRequestId0 + 11);
+
+            var expectedFirstMessage = GetExpectedSaslStartMessage(actualRequestId0, expectedClientFirstMessage);
+            var expectedSecondMessage = GetExpectedSaslContinueMessage(actualRequestId1, expectedClientSecondMessage);
+
+            sentMessages[0].Should().Be(expectedFirstMessage);
+            sentMessages[1].Should().Be(expectedSecondMessage);
+        }
+
+        private static string GetExpectedSaslContinueMessage(int requestId, BsonDocument clientMessage)
+        {
+            return  "{" +
+                        "opcode: \"query\", " +
+                        $"requestId: {requestId}, " +
+                        "database: \"$external\", " +
+                        "collection: \"$cmd\", " +
+                        "batchSize: -1, " +
+                        "slaveOk: true, " +
+                        "query: " +
+                        "{ " +
+                            "\"saslContinue\" : 1, " +
+                            "\"conversationId\" : 1, " +
+                            $"\"payload\" : new BinData(0, \"{ToBase64(ToBytes(clientMessage))}\") " +
+                        "}" +
+                    "}";
```

Comment:
We never indent more than one tab stop (4 spaces) at a time. I would format like this:

    return 
        "{" +
            "opcode: \"query\", " +
            $"requestId: {requestId}, " +
            "database: \"$external\", " +
            "collection: \"$cmd\", " +
            "batchSize: -1, " +
            "slaveOk: true, " +
            "query: " +
            "{ " +
                "\"saslContinue\" : 1, " +
                "\"conversationId\" : 1, " +
                $"\"payload\" : new BinData(0, \"{ToBase64(ToBytes(clientMessage))}\") " +
            "}" +
        "}";